project-icp/bee-pollinator-app
# Add Splash, Sidebar component

## Overview

Adds a Splash component with basic styling. Extracts Map component into its own placeholder. Adjusts Prototype component to include only the sidebar contents for now. Adjusts App component accordingly.

Connects #297 

### Demo

![image](https://user-images.githubusercontent.com/1430060/47672746-24405b00-db89-11e8-87ac-8e1d08b71942.png)

Tagging @jfrankl 	for visual and style review.

### Notes

#### BEM

Am I following BEM correctly for naming CSS classes?

```html
<div className="splash__info_box">
    <div className="splash__info_box__title">
        Get Started
    </div>
    <div className="splash__info_box__body">
        Click on the map to see how it stacks up
    </div>
</div>
```

It feels a little redundant having to specify `splash__info_box` multiple times.

#### `stylelint`

I liked that `stylelint` will auto-format the SCSS files on save, including the order of CSS rules, as long as `beekeepers start` is running.

#### Mouse Cursor Icon

I did not see an icon or image in the prototype for the mouse cursor click icon shown in the mockup:

![image](https://user-images.githubusercontent.com/1430060/47241463-be4e1980-d3b9-11e8-9c6c-c25b207cf7ba.png)

Any recommendations on where to find it?

#### Style Variable Use

I got the page to be as close to the prototype as I could, but it's not quite identical yet in terms of text size, and it was taking me too long to tune the details, so I've left it as a design task for later. One thing that I did do, however, is use the existing type size variables wherever I could:

```sass
&__header {
    margin-top: 1rem;
    margin-bottom: $font-size-xl;
    font-weight: 900;
    font-size: $font-size-xxl;
    line-height: $font-size-3xl;
}
```

even in places that aren't strictly type related, such as `margin-bottom` above. This was done in an effort to maintain the established rhythm, but if this is seen as bad practice I can change those out to be literal values.

## Testing Instructions

* Check out this branch and `./scripts/beekeepers.sh start`
* Go to [:8000/?beekeepers](http://localhost:8000/?beekeepers) and ensure you see the splash screen
